common-content/en/module/tools/installing-npm-dependencies/index.md

```diff
@@ -0,0 +1,124 @@
+
+program.parse();
+
+const argv = program.args;
+if (argv.length != 1) {
+    console.error(`Expected exactly 1 argument (a path) to be passed but got ${argv.length}.`);
+    process.exit(1);
+}
+const path = argv[0];
+const char = program.opts().char;
+
+const content = await fs.readFile(path, "utf-8");
+const wordsContainingChar = content.split(" ").filter((word) => word.indexOf(char) > -1).length;
+console.log(wordsContainingChar);
+```
+
+{{<note type="Exercise">}}
+Try running this program with the `--help` flag.
+
+What do you see? Where do you think this behaviour and text came from?
+
+We didn't have to write all the code for this functionality - `commander` did most of it for us.
+{{</note>}}
+
+{{<note type="Exercise">}}
+Try running the program with different values of the `-c` flag. Try also specifying some other flags, like `--count`.
+
+Make sure you understand how it's behaving, and why.
+{{</note>}}
+
+Let's run through what we changed:
+
+```js
+program
+    .name("count-containing-words")
+    .description("Counts words in a file that contain a particular character")
+    .option("-c, --char <char>", "The character to search for", "-");
+```
+
+We told `commander` information about our program. We gave it a name, a description, and told it that it should allow a user to pass a flag name `-c` (or equivalently `--char`), and use a default value of `-` for that flag if it's not specified.
+
+```js
+program.parse();
+```
+
+We asked `commander` to interpret the command line arguments our program was given, based on what options we wanted to allow. If it sees something it doesn't understand, it will error.
+
+```js
+const argv = program.args;
+```
+
+Instead of asking NodeJS's process module for all of the program's arguments, we're asking `commander` to tell us "after you understood and removed all the flags, what arguments were left?"
+
+Then our `if` check about the number of arguments is exactly the same as before.
+
+```js
+const char = program.opts().char;
+```
+
+We are getting the `char` flag that `commander` interpreted and storing it in a variable.
+
+```js
+const wordsContainingChar = content.split(" ").filter((word) => word.indexOf(char) > -1).length;
+console.log(wordsContainingChar);
+```
+
+We have renamed our `wordsContainingHyphens` variable to `wordsContainingChar` because we're no longer always looking for hyphens, and changed the `indexOf` call to look for the value of the `char` variable instead of always a `-`.
+
+We only needed to make a few small changes to get all of this new functionality:
+* Support for accepting a new command line flag.
+* `--help` support explaining how to use the program.
+* Detection for if someone passes flags that aren't known, and warning them about this (and even suggesting what they maybe meant).
+
+We could have written all of this code ourselves. But using a library meant we could focus on what's unique about our problem, rather than spending time implementing flag parsing.
```

Comment:
Don't 100% love this wording, but I think it is worth calling this out?

```suggestion
We could have written all of this code ourselves. But using a library meant we could focus on what's unique about our problem, rather than spending time implementing flag parsing.

This is a very common task in software development in the real world, joining together libraries (written by other people) to create some new unique solution.
```